ppl/zqd/middleware.go

```diff
@@ -25,15 +23,14 @@ func getRequestID(ctx context.Context) string {
 // context. If the header "X-Request-ID" exists this will be used, otherwise
 // one will be generated.
 func requestIDMiddleware() mux.MiddlewareFunc {
-	var count int64
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			reqID := r.Header.Get(requestIDKey)
+			reqID := r.Header.Get(api.RequestIDHeader)
 			if reqID == "" {
-				reqID = strconv.FormatInt(atomic.AddInt64(&count, 1), 10)
+				reqID = ksuid.New().String()
 			}
-			w.Header().Add(requestIDKey, reqID)
-			ctx := context.WithValue(r.Context(), requestIDKey, reqID)
```

Comment:
I have a question for background: the previous request id was a sequence number specific to this zqd process, but the new request id is a ksuid. I assume you are making this change so request id can be used to correlate messages from multiple zqd instances, right?

Reply:
Yep. Before I was just using increasing ints starting from 0 which would result in request_id collisions between instances.